File: Class/Practise 4.1/src/IngredientsReader.cpp
#include "IngredientsReader.h"
#include <print>
#include <fstream>
#include <sstream>
#include <iostream>

std::unordered_map<std::string, Ingredient> IngredientsReader::readIngredients(const std::string& filename) {
    std::unordered_map<std::string, Ingredient> ingredients;
    std::ifstream file(filename);
    if (!file) {
        std::println(stderr, "Unable to open file: {}", filename);
        return ingredients;
    }

    std::string line;
    while (std::getline(file, line)) {
        std::istringstream iss(line);
        std::string name;
        double cost;
        if (iss >> name >> cost) {
            ingredients[name] = Ingredient(name, cost);
        } else {
            std::println(stderr, "Invalid line format: {}", line);
        }
    }
    return ingredients;
}

File: Home/Home 4.1/IngredientsReader.swift
import Foundation

class IngredientsReader {
    func readIngredients(from filename: String) -> [String: Ingredient] {
        var ingredients: [String: Ingredient] = [:]
        guard let fileURL = Bundle.main.url(forResource: filename, withExtension: "txt"),
              let fileContents = try? String(contentsOf: fileURL) else {
            print("Unable to open file: \(filename)")
            return ingredients
        }

        let lines = fileContents.split(separator: "\n")
        for line in lines {
            let parts = line.split(separator: " ")
            if parts.count == 2, let cost = Double(parts[1]) {
                let name = String(parts[0])
                ingredients[name] = Ingredient(name: name, cost: cost)
            } else {
                print("Invalid line format: \(line)")
            }
        }
        return ingredients
    }
}
